hdfs/src/main/java/com/ibm/crail/hdfs/CrailHDFSUtils.java

```diff
@@ -0,0 +1,19 @@
+package com.ibm.crail.hdfs;
+
+import java.util.Iterator;
+
+import com.ibm.crail.CrailNode;
+
+public class CrailHDFSUtils {
+	public static Iterator<String> getIterator(CrailNode node) throws Exception {
+		if (node.getType().isDirectory()){
```

Comment:
I understand that having the types registered is major rework for a different pull request but can we not at least have something like node.getType().isContainer() -> return node.asContainer().listEntries() ?